### ISSUE TITLE
Fails on some perlbal tunables

### ISSUE THREAD

**Issue by eserte** [xsawyerx/perlbal-plugin-sessionaffinity#1]
It seems that there are perlbal tunables which cause the sessionaffinity plugin to fail. Trying to use sessionaffinity in a perlbal system where also the throttle plugin is in use fails like this:
```
crash log: Attempt to access disallowed key 'log_only' in a restricted hash at /usr/share/perl5/Perlbal/Plugin/SessionAffinity.pm line 248.
```
My first attempt was to skip this key, but there were more: `log_events`, `throttle_threshold_seconds`... so I ended putting an `eval { }` around the `defined` call in this line.

Next problem was a stack trace starting with:
```crash log: MC:err: Error defining trusted upstream proxies: could not parse ARRAY(0x33d0cd8) at /usr/share/perl5/Perlbal/ManageCommand.pm line 42.```
Indeed, `trusted_upstream_proxies` is defined in the service definition where the sessionaffinity plugin was added. My quick fix here was also an additional skip.

**Comment by eserte** [comment]
A possible workaround for the trusted_upstream_proxies problem seems like this:
```
                if ( $role eq '*' || $role eq $svc_role ) {
                    my $val = $svc->{$tunable_name};
                    if (ref $val eq 'ARRAY') {
                        $val = join(",", @$val);
                    }
                    $nodeservice->set( $tunable_name, $val );
                }
```
At least this works for `trusted_upstream_proxies`: its value is expected to be a comma-separated list in the configuration, but will be turned into an array within the Service object.

There seem to be other configuration parameters with non-scalar values; I did not check them. Some seem to have a dumper function which maybe could be misused as an "inverse" function for getting a propert value for set().

**Comment by xsawyerx** [comment]
I just released a new version (0.110) that should fix both of these cases. (I've had this tab open since you opened the issue and only now found some time to work on it. Sorry.)

* To handle the disallowed key, I now check with `exists` before calling `defined`. This seemed to work when I tested the locking mechanism. This happens when the hash is locked but the key does not exist, so the `exists` should solve it. If this continued, please let me know.
* To handle the arrayref issue, I took your patch to simply merge the values when it's an arrayref.

Thanks for opening this issue and for the suggested fix. I credited you in the Changes file as well.
